### DISCUSSION TITLE
Cloudflare Tunnel And Mobile App

### DISCUSSION THREAD

**Discussion by unnerving-sprinkler** [immich-app/immich#6184] [Q&A]
Hoping someone here can point me in the right direction here. 

I am running a Cloudflare zero trust tunnel for my Immich server. 

When I have no authentication setup for the tunnel e.g. immich.mydomain.com, the app works absolutely fine. 

When I setup an application to secure my connection which is setup as Subdomain: immich.mydomain.com Policy: Allow Email "My Email" Auth: Either through GitHub Or One Time email, I can still access immich through a browser immich.mydomain.com and doing my one time email, but when I try and access through the app, I know it does talk at least some because I get past the enter a URL and into the login page, but when I attempt to login I get "error logging you in, please check server URL, username, and password"

**Comment by bo0tzz** [comment]
The Immich app doesn't support external proxy auth like this. See #1604

**Reply by smeingast** [reply]
Ah, too bad. I just came across the same issue after setting up authentication through the cloudflare tunnel. :/

**Reply by unnerving-sprinkler** [reply]
I suspect there's a way around this using the cloud flare zero trust VPN on your device, but I didn't go far down that rabbit hole. Easier to just setup tailscsle.

**Reply by smeingast** [reply]
I am also using tailscale, but they do not support subdomains with MagicDNS and every one of my attempts to set up SSL connections to various services running on a single device failed in the end. It's either a problem with certificates or DNS and at this point I just hope that they will support subdomains in the near future.

**Comment by reefland** [comment]
The [Immich FAQ](https://immich.app/docs/FAQ/#why-are-only-photos-and-not-videos-being-uploaded-to-immich) makes reference to Cloudflare Tunnel so I assumed it was supported, until I tried it... not clear if FAQ is relevant to Mobile App, but the top of the page starts with Mobile App.

![image](https://github.com/immich-app/immich/assets/16230914/12ece8cb-e1c5-4887-8732-8081d2c1979c)

**Reply by unnerving-sprinkler** [reply]
Cloudflare Tunnel absolutely works when setup without authentication, it's the "zero trust" login page that causes issues. 

I have just been avoiding the issue for now, but I intend on setting up some good geo-blocking rules and just letting it roll.

**Reply by reefland** [reply]
Ah, got ya... didn't try without authentication.

**Reply by nsantini** [reply]
Hi @unnerving-sprinkler , I have set up Immich behind a Zero Trust tunnel with Nginx in the host machine. But I can't upload images from the mobile app. Even tho I set up the client_max_body_size as recommended.
Did you have to do anything else besides the standard installation to make it work?
Thanks
